autopush/router/gcm.py

```diff
@@ -47,6 +47,11 @@ def route_notification(self, notification, uaid_data):
 
     def _route(self, notification, router_data):
         """Blocking GCM call to route the notification"""
+        if not self.gcm:
+            self.gcm = gcmclient.GCM(
+                router_data.get("creds",
+                                {}).get("auth",
+                                        self._default_auth))
```

Comment:
If this really needs to be done in the thread-pool, it should be wrapped with a:

```python
with self._gcm_lock:
    if not self.gcm:
....
```